pkg/daemon/rpm-ostree.go

```diff
@@ -0,0 +1,73 @@
+package daemon
+
+import (
+	"encoding/json"
+	"fmt"
+	utils "github.com/openshift/machine-config-operator/pkg/utils"
+	"strings"
+)
+
+// Subset of `rpm-ostree status --json`
+// https://github.com/projectatomic/rpm-ostree/blob/bce966a9812df141d38e3290f845171ec745aa4e/src/daemon/rpmostreed-deployment-utils.c#L227
+type RpmOstreeState struct {
+	Deployments []RpmOstreeDeployment
+}
+
+type RpmOstreeDeployment struct {
+	Id           string   `json:"id"`
+	OSName       string   `json:"osname"`
+	Serial       int32    `json:"serial"`
+	Checksum     string   `json:"checksum"`
+	Version      string   `json:"version"`
+	Timestamp    uint64   `json:"timestamp"`
+	Booted       bool     `json:"booted"`
+	Origin       string   `json:"origin"`
+	CustomOrigin []string `json:"custom-origin"`
+}
+
+func getBootedDeployment() (*RpmOstreeDeployment, error) {
+	var rosState RpmOstreeState
+	output, err := utils.RunGetOut("rpm-ostree", "status", "--json")
+	if err != nil {
+		return nil, err
+	}
+
+	if err := json.Unmarshal(output, &rosState); err != nil {
+		return nil, fmt.Errorf("Failed to parse `rpm-ostree status --json` output: %v", err)
+	}
+
+	// just make it a hard error if we somehow don't have any deployments
+	if len(rosState.Deployments) == 0 {
+		return nil, fmt.Errorf("Not currently booted in a deployment")
```

Comment:
since this is entirely the same error as the one at the end of the function, and the one at the end of the function will be hit if the length is zero and this isn't here, do we need this if statement at all?

Reply:
Oh yeah, I meant to drop this but forgot!